### DISCUSSION TITLE
Feature Request: More Versatile Web Search

### DISCUSSION THREAD

**Discussion by Thraxbert** [anyrun-org/anyrun#122] [Ideas]
Anyrun is, far and away, my favorite launcher on Linux. I've been a fulltime Linux user for two years and went through a number of distro+launcher combos until finally setting on Arch+Hyprland+Anyrun and I could scarcely be happier.

But there is one thing I miss from Windows: the web search implementation in the launcher "Launchy."

It allows you to define a shortcut, URL, and wildcard for _any_ search engine in its web runner configuration. You can have multiple search engines configured based on different shortcuts. If I were to rewrite the config file to put it in "Anyrun" terms to simultaneously support Bing, Google, and Amazon, it would look something like this:

```
Config(
	Google(
	name: "Google",
	url: "https://www.google.com/search?q={}",
	prefix: g
	)

	Amazon(
	name: "Amazon",
	url: "https://www.amazon.com/s?k={}",
	prefix: am
	)

	Bing(
	name: "Bing",
	url: "https://www.bing.com/search?q={}",
	prefix: b
	)
)
```

In other words: the websearch engine would be written to be more of a "webrunner" engine, that could support any URL conformant with the {} wildcard syntax that is currently available. 

Thank you for making such a wonderful program and considering my request.

**Comment by kuokuo123** [comment]
I've written a plugin recently to realize this function. Take a look and see if it suits your need!

https://github.com/kuokuo123/anyrun-websearch-plus

**Reply by Thraxbert** [reply]
This is perfect! Thank you so much! Your implementation is great. This is the very last thing that I missed from Windows, and now it's GONE! You are the best.
